src/repositories/organisations/inmemory.js

```diff
@@ -19,6 +19,10 @@ export const createInMemoryOrganisationsRepository = (
     async findAll() {
       // Return a deep clone to prevent consumers mutating internal state.
       return structuredClone(storage)
+    },
+    async findByOrgId(orgId) {
+      const found = storage.find((o) => o.orgId === orgId)
```

Comment:
nitpick: could use a Map as the backing store so you can look up by id

Reply:
I'd argue that the in-memory adapter should not attempt to replicate MongoDB in any way. The port is there to abstract away the MongoDB-specific stuff, not so we can switch out MongoDB for anything else but to make sure it is designed in the best possible way for the application to consume, regardless of the implementation.
If we make other adapters behave like MongoDB then we are in danger of leaking MongoDB behaviour into the application, rather than the application defining what it ideally needs via the port.

Reply:
Like I said it's a nitpick tho 😆